### PR TITLE
Implements #807

### PR DESCRIPTION
**Pull request by Bart1909**
Hi @jonaswinkler,
I've implemented #807 , or at least, I tried :-)

If you open /asn/{asn} it will search the document with the asn and redirect you to the document page.
If no document is found, you're redirected to 404.

Hope, this is fine for you.

Best wishes

Tobi

### CODE REVIEW

**Comment by shamoon** [comment]
Nice. Though probably the component should be just `DocumentASNComponent` not `DocumentASNComponentComponent` and moved to `/components/document-asn/document-asn.component`

**Comment by shamoon** [comment]
Also you'll want to change the branch from `master` to `dev` as thats where changes get merged before release

**Comment by Bart1909** [comment]
Renamed the component and changed the target branch to dev.
I cannot change the source branch. If this is necessary, I will create a new PR

**Comment by jonaswinkler** [comment]
> I cannot change the source branch.

If you're worried about the changes to the readme and such, no worries. That's totally fine.

Otherwise, looks good.

I'm quite busy these days. I'll dedicate a day on easter holidays to paperless, there's quite a few things I need to catch up on and get this released.

File: src-ui/src/app/components/document-asn/document-asn.component.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<p>Searching document with asn {{asn}}</p>
```
Comment (jonaswinkler) [inline src-ui/src/app/components/document-asn/document-asn.component.html:1]: Change to

```
<p i18n>Searching for document with ASN {{asn}}...</p>
```

and execute `ng xi18n --ivy` after that.

File: src-ui/src/app/components/document-asn/document-asn.component.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  styleUrls: ['./document-asn.component.scss']
+})
+export class DocumentAsnComponent implements OnInit {
+
+  asn: string;
+  constructor(
+    private documentsService: DocumentService,
+    private route: ActivatedRoute,
+    private correspondentService: CorrespondentService,
+    private documentTypeService: DocumentTypeService,
```
Comment (jonaswinkler) [inline src-ui/src/app/components/document-asn/document-asn.component.ts:20]: Please remove services you don't need.

**Comment by Bart1909** [comment]
Okay, very fine :-)

It's fine too, i this will released after easter.
If you accept the path /asn/{asn} I can start printing my stickers :-)

Best wishes

File: src-ui/src/app/components/document-asn/document-asn.component.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import {FILTER_ASN} from "../../data/filter-rule-type";
+
+@Component({
+  selector: 'app-document-asncomponent',
+  templateUrl: './document-asn.component.html',
+  styleUrls: ['./document-asn.component.scss']
+})
+export class DocumentAsnComponent implements OnInit {
+
+  asn: string;
```
Comment (jonaswinkler) [inline src-ui/src/app/components/document-asn/document-asn.component.ts:15]: No `;` required in typescript.

**Pull request merged**